builtInFunctions/baseGuardAccount.go

```diff
@@ -31,7 +37,25 @@ func (bfa *baseGuardAccount) checkGuardAccountArgs(
 	acntSnd vmcommon.UserAccountHandler,
 	vmInput *vmcommon.ContractCallInput,
 ) error {
-	err := bfa.checkBaseAccountGuarderArgs(acntSnd, vmInput, noOfArgsGuardAccount)
+	if check.IfNil(acntSnd) {
+		return fmt.Errorf("%w for sender", ErrNilUserAccount)
+	}
+	if vmInput == nil {
+		return ErrNilVmInput
+	}
+
+	senderAddr := acntSnd.AddressBytes()
+	senderIsNotCaller := !bytes.Equal(senderAddr, vmInput.CallerAddr)
```

Comment:
this cannot happen. Like this is ensured from how the built in function is called. BlockChainHook.getAccount reads from trie the accounts for acntSnd == vmInput.Caller and acntDst == vmInput.RecvAddress

Reply:
yes, but it is still harmless, just wanted to make sure if something changes, this still holds.